Doc/reference/expressions.rst

```diff
@@ -639,6 +639,13 @@ after resuming depends on the method which resumed the execution.  If
 :meth:`~agen.asend` is used, then the result will be the value passed in to
 that method.
 
+If an asynchronous generator is not resumed before it reaches the end of function
+but the event loop where the generator is bound to continues to run (e.g., when
+the caller task is cancelled), the caller must explicitly close the async
+generator by calling :meth:`~agen.aclose` method to free the resources allocated
+in the async generator function's stack and detach the generator from the event
+loop.
+
```

Comment:
It could be explained more clearly-- most readers won't have in mind how this situation comes up.  And the rationale could be more detailed.

> If an asynchronous generator happens to exit early by :keyword:`break`, the caller task being cancelled, or other exception, the generator's async cleanup code will run and possibly raise exceptions or access context variables in an unexpected context-- perhaps after the lifetime of tasks it depends on or the event loop itself.  To prevent this, the caller must explicitly close the async generator by calling :meth:`~agen.aclose` method to finalize the generator and ultimately detach it from the event loop.

Reply:
Subtlety: It can't happen after the event loop exits, thanks to the async gen gc hooks.

Reply:
Updated!